examples/stdlib/bytearray/bytearray_compare.effekt

```diff
@@ -0,0 +1,14 @@
+module examples/pos/bytearray_compare
+
+import bytearray
+
+def main() = {
+  val b1 = fromString("Hey")
+  val b2 = fromString("Hey")
+  val b3 = fromString("He")
+  val b4 = fromString("Heys")
+  println(compareByteArray(b1, b1))
+  println(compareByteArray(b1, b2))
+  println(compareByteArray(b1, b3))
+  println(compareByteArray(b1, b4))
```

Comment:
Can you make sure that these test cases cover all paths in the implementation? (shorter, longer,  different prefix, etc.)

Reply:
FWIW, I'd love a test comparing [haha!] this with equality. :)

```
compare == 0 <=> equals == true
compare != 0 <=> equals == false
```